packages/structure/src/lib.rs

```diff
@@ -0,0 +1,237 @@
+use serde_json::{from_str, json, Value};
+use std::env;
+use TOCEntryType::*;
+
+#[macro_export]
+macro_rules! inline_content {
+    ($expr:expr) => {
+        json!({
+            "name": "inline_content",
+            "data": $expr
+        })
+    }
+}
+
+pub struct StructureCounter {
+    headings: Vec<u64>,
+    figures: u64,
+    tables: u64,
+}
+
+#[derive(PartialEq)]
+enum TOCEntryType {
```

Comment:
I think its more common to still using PascalCase for abbreviations like TOC. But not that important! : )
```suggestion
enum TocEntryType {
```